src/pages/Articles.jsx

```diff
@@ -23,7 +23,9 @@ const Articles = () => {
     } else setQueryString("");
   };
 
+  
   useEffect(() => {
+    setSearchParams(queryString);
     const fetchArticles = async () => {
       const response = await getArticles(queryString);
```

Comment:
You should set your loading to false by default , then true in here, then finally false again to better represent what is happening